src/headless/converse-roster.js

```diff
@@ -88,10 +88,14 @@ converse.plugins.add('converse-roster', {
                 }
             } else {
                 try {
-                    await _converse.rostergroups.fetchRosterGroups().then(() => {
-                        _converse.emit('rosterGroupsFetched');
-                        return _converse.roster.fetchRosterContacts();
-                    });
+                    /* Make sure not to run fetchRosterContacts async, since we need
```

Comment:
eee

Reply:
One way to test for this, is to check that `rosterGroupsFetched` is called *before* `rosterContacts` fetched.

I'm happy to help you with writing such a test.

Reply:
Sure, I'll have a go at that this week. :)